.github/workflows/main.yml

```diff
@@ -40,16 +40,45 @@ jobs:
         with:
           fetch-depth: 0 # checkout tags so that dynver works properly (we need the version for MiMa)
       - name: Set up JDK 11
-        uses: actions/setup-java@v1
+        uses: coursier/setup-action@v1.1.2
         with:
-          java-version: 11
+          jvm: adopt:11
       - name: Cache sbt
         uses: coursier/cache-action@v6
         with:
           extraKey: sbt-cache-${{ runner.os }}
       - name: Check MiMa # disable for major releases
         run: sbt -v core/mimaReportBinaryIssues
 
+  sourcegraph:
+    name: Upload index to sourcegraph
+    needs: [ci]
+    # run on external PRs, but not on internal PRs since those will be run by push to branch
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-20.04
+    env:
+      JAVA_OPTS: -Xmx4G
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: coursier/setup-action@v1.1.2
+        with:
+          jvm: adopt:11
+          apps: lsif-java
+      - name: Cache sbt
+        uses: coursier/cache-action@v6
+        with:
+          extraKey: sbt-cache-${{ runner.os }}
+      - name: Generate LSIF
+        run: lsif-java index
+      - name: Upload LSIF data
+        uses: sourcegraph/lsif-upload-action@master
+        with:
+          endpoint: https://sourcegraph.com
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          file: dump.lsif
```

Comment:
don't we need to tag the upload with the version or sth like this?

Reply:
You mean magnolia version or scala version?
1. magnolia version - The lsif upload will be associated with given commit hash, so no.
2. scala version - Sourcegraph for now doesn't support cross-compiled projects. However, magnolia is kind of special as different branches build different versions.

Reply:
no, I meant the magnolia version for which we are sending the source info :)

Reply:
or is that in the dump file?

Reply:
No it is not in the dump file as far as i know. The lsif upload will be associated with given commit hash.

Reply:
> The lsif upload will be associated with given commit hash.

This is at least what I would expect it to do, but it is only assumption.

Reply:
That is correct, you can check it at https://sourcegraph.com/github.com/softwaremill/magnolia/-/code-intelligence/uploads?visible=1

```
[github.com/softwaremill/magnolia](https://sourcegraph.com/github.com/softwaremill/magnolia)
Directory [/](https://sourcegraph.com/github.com/softwaremill/magnolia@f8c53577718c717b76fb625d8f47d69fb71a7f3f) 
indexed at commit [f8c5357](https://sourcegraph.com/github.com/softwaremill/magnolia/-/commit/f8c53577718c717b76fb625d8f47d69fb71a7f3f) by lsif-java
Completed 1 minute ago
```